pkg/api/norman/customization/aks/handler.go

```diff
@@ -0,0 +1,232 @@
+	ns, name := ref.Parse(credID)
+	if ns == "" || name == "" {
+		logrus.Debugf("[AKS] invalid cloud credential ID %s", credID)
+		return http.StatusBadRequest, fmt.Errorf("invalid cloud credential ID %s", credID)
+	}
+
+	var accessCred client.CloudCredential //var to check access
+	if err := access.ByID(h.generateAPIContext(req), &schema.Version, client.CloudCredentialType, credID, &accessCred); err != nil {
+		apiError, ok := err.(*httperror.APIError)
+		if !ok {
+			return httperror.NotFound.Status, err
+		}
+		if apiError.Code.Status == httperror.NotFound.Status {
+			return httperror.InvalidBodyContent.Status, fmt.Errorf("cloud credential not found")
+		}
+		if apiError.Code.Status != httperror.PermissionDenied.Status {
+			return httperror.InvalidBodyContent.Status, err
+		}
+		var clusterID string
+		if clusterID = req.URL.Query().Get("clusterID"); clusterID == "" {
+			return httperror.InvalidBodyContent.Status, fmt.Errorf("cloud credential not found")
+		}
+		if errCode, err := h.clusterCheck(h.generateAPIContext(req), clusterID, credID); err != nil {
+			return errCode, err
+		}
+	}
+
+	cc, err := h.secretsLister.Get(ns, name)
+	if err != nil {
+		logrus.Debugf("[AKS] error accessing cloud credential %s", credID)
+		return httperror.InvalidBodyContent.Status, fmt.Errorf("error accessing cloud credential %s", credID)
+	}
+	cap.SubscriptionID = string(cc.Data["azurecredentialConfig-subscriptionId"])
+	cap.ClientID = string(cc.Data["azurecredentialConfig-clientId"])
+	cap.ClientSecret = string(cc.Data["azurecredentialConfig-clientSecret"])
+
+	cap.TenantID = req.URL.Query().Get("tenantId")
+	if cap.TenantID == "" {
+		logrus.Debugf("[AKS] error getting tenantId")
+		return http.StatusBadRequest, fmt.Errorf("error getting tenantId")
+	}
+
+	cap.BaseURL = req.URL.Query().Get("baseUrl")
+	if cap.BaseURL == "" {
+		cap.BaseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+	cap.AuthBaseURL = req.URL.Query().Get("authBaseUrl")
+	if cap.AuthBaseURL == "" {
+		cap.AuthBaseURL = azure.PublicCloud.ActiveDirectoryEndpoint
+	}
+	cap.ResourceLocation = req.URL.Query().Get("region")
+
+	return http.StatusOK, nil
+}
+
+func (h *handler) clusterCheck(apiContext *types.APIContext, clusterID, cloudCredentialID string) (int, error) {
+	clusterInfo := map[string]interface{}{
+		"id": clusterID,
+	}
+
+	clusterSchema := h.schemas.Schema(&mgmtSchema.Version, client.ClusterType)
+	if err := h.ac.CanDo(v3.ClusterGroupVersionKind.Group, v3.ClusterResource.Name, "update", apiContext, clusterInfo, clusterSchema); err != nil {
+		return httperror.InvalidBodyContent.Status, fmt.Errorf("cluster not found")
+	}
+
+	cluster, err := h.clusterLister.Get("", clusterID)
+	if err != nil {
+		if httperror.IsNotFound(err) {
+			return httperror.InvalidBodyContent.Status, fmt.Errorf("cluster not found")
+		}
+		return httperror.ServerError.Status, err
+	}
+
+	if cluster.Spec.GKEConfig.GoogleCredentialSecret != cloudCredentialID {
```

Comment:
```suggestion
	if cluster.Spec.AKSConfig.AzureCredentialSecret != cloudCredentialID {
```

Reply:
I fixed it, but it will require to merge https://github.com/rancher/rancher/pull/31333 first